### PR TITLE
Badge: Update references to point to new package component

### PR DESCRIPTION
**Pull request by sixhours**
Related to #77564

## Proposed Changes

* Following up on #78809 and #79303, this repoints any references to `Badge` from Calypso components to `@automattic/components`
* I recognize this is a sizable change that touches many files. We effectively made this change in the background with #79303, so all Badge components are running the new/relocated code already. This PR repoints the reference from the wrapper to the "real" code, so I don't expect this to cause any issues, and I've spot-checked as many instances as I could find on the front end.

## Testing Instructions

* Make sure tests pass.
* Do a spot-check of Badge instances to ensure they still look/work as expected.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by JessBoctor — APPROVED** [review]
LGTM. I checked the domain connection page and I didn't see anything that was off. 

I also re-ran the TC tests and they are all passing now.

**Review by bogiii — APPROVED** [review]

**Review by markbiek — APPROVED** [review]
LGTM!

File: client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx [hunk 1]
```diff
@@ -220,6 +220,7 @@ export default function CampaignItemDetails( props: Props ) {
 			label: __( 'Contact support' ),
 			onClick: async () => {
 				setShowErrorDialog( false );
+				// eslint-disable-next-line wpcalypso/i18n-unlocalized-url
 				window.open( 'https://wordpress.com/support/', '_blank' );
```
Comment (noahtallen) [inline client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:224]: by the way, you can use `import { useLocalizeUrl } from '@automattic/i18n-utils';` and then `useLocalizeUrl( 'https://wordpress.com/support/' )` to resolve this issue. Was there a different reason for using disable-line instead?

Comment (sixhours) [reply client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:224]: TBH I didn't want to overcomplicate this PR (more than it was already) and dig into it at the time. 😅 Happy to start a new PR to take care of these, though!

Comment (sixhours) [reply client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:224]: @noahtallen Thanks for flagging this, I've opened https://github.com/Automattic/wp-calypso/pull/79627 to follow up!

Comment (noahtallen) [reply client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:224]: I think it's a newer rule, so it's easy to miss!

**Pull request merged**